packages/web3torrent/src/contexts/omni-provider.tsx

```diff
@@ -0,0 +1,21 @@
+import {BudgetContext, useBudgetContext} from './budget-context';
+import {InitializationContext, useInitializationContext} from './initialization-context';
+import React from 'react';
+import {ChannelContext, useChannelContext} from './channel-context';
+import {TorrentContext, useTorrentContext} from './torrent-context';
+
+export const OmniProvider: React.FC = ({children}) => {
+  const initializationContext = useInitializationContext();
+  // TODO: Probably a more elegant way of doing this
+  return (
+    <InitializationContext.Provider value={initializationContext}>
```

Comment:
The idea behind constructing a `InitializationContext` and passing it to other contexts is that other contexts can initialize the "wallet" if needed.

This means that if the UI calls `useContext(BudgetContext)` the budget context can use the initialize context to intialize the "wallet" so the UI doesn't have to care about whether `initialize` has been called or not.

Reply:
My comment is slightly misleading as it makes it sound that `initialize` will occur when the context is accessed. However the `initialize` will be called at startup (see https://github.com/statechannels/monorepo/pull/1528#issuecomment-618591176)